README.md

```diff
@@ -1,169 +1,171 @@
-```
-- uses: actions/download-artifact@v2
+```yaml
+steps:
+- uses: actions/download-artifact@v4
   with:
     name: my-artifact
-    path: my-artifact
+    path: your/destination/dir
+- name: Display structure of downloaded files
+  run: ls -R your/destination/dir
 ```
 
 
-# Download All Artifacts
+### Download All Artifacts
 
 If the `name` input parameter is not provided, all artifacts will be downloaded. **To differentiate between downloaded artifacts, a directory denoted by the artifacts name will be created for each individual artifact.**
+
 Example, if there are two artifacts `Artifact-A` and `Artifact-B`, and the directory is `etc/usr/artifacts/`, the directory structure will look like this:
+
 ```
-  etc/usr/artifacts/
-      Artifact-A/
-          ... contents of Artifact-A
-      Artifact-B/
-          ... contents of Artifact-B
+etc/usr/artifacts/
+    Artifact-A/
+        ... contents of Artifact-A
+    Artifact-B/
+        ... contents of Artifact-B
 ```
 
-Download all artifacts to a specific directory
+Download all artifacts to the current working directory:
+
 ```yaml
 steps:
-- uses: actions/checkout@v3
-
-- uses: actions/download-artifact@v3
-  with:
-    path: path/to/artifacts
-    
+- uses: actions/download-artifact@v4
 - name: Display structure of downloaded files
   run: ls -R
-  working-directory: path/to/artifacts
 ```
 
-Download all artifacts to the current working directory
+Download all artifacts to a specific directory:
+
 ```yaml
 steps:
-- uses: actions/checkout@v3
-
-- uses: actions/download-artifact@v3
-
+- uses: actions/download-artifact@v4
+  with:
+    path: path/to/artifacts
 - name: Display structure of downloaded files
-  run: ls -R
+  run: ls -R path/to/artifacts
 ```
 
-# Download path output
+### Download Artifacts from other Workflow Runs or Repositories
 
-The `download-path` step output contains information regarding where the artifact was downloaded to. This output can be used for a variety of purposes such as logging or as input to other actions. Be aware of the extra directory that is created if downloading all artifacts (no name specified).
+It may be useful to download Artifacts from other workflow runs, or even other repositories. By default, the permissions are scoped so they can only download Artifacts within the current workflow run. To elevate permissions for this scenario, you can specify a `github-token` along with other repository and run identifiers:
 
 ```yaml
 steps:
-- uses: actions/checkout@v3
-
-- uses: actions/download-artifact@v3
-  id: download
+- uses: actions/download-artifact@v4
   with:
-    name: 'my-artifact'
-    path: path/to/artifacts
-
-- name: 'Echo download path'
-  run: echo ${{steps.download.outputs.download-path}}
+    name: my-other-artifact
+    github-token: ${{ secrets.GH_PAT }} # token with actions:read permissions on target repo
```

Comment:
I wonder if it might be better to provide an example using the `GITHUB_TOKEN` as well:

```
...
permissions:
      actions: read
...
with:
...
  github-token ${{ secrets.GITHUB_TOKEN }}
...
```

Reply:
🤔 I'm not sure how useful that would be for a customer scenario. That permission would only grant the scope for the current run, which is the same as not using a GITHUB_TOKEN at all (which would be using the internal calls)

Reply:
Fair!